src/NKSolver/NKSolvers.F90

```diff
@@ -3166,13 +3220,24 @@ subroutine physicalityCheckANK(lambdaP)
     call EChk(ierr,__FILE__,__LINE__)
 
     ! Make sure that we did not get any NaN's in the process
-    if (isnan(lambdaL)) lambdaL = zero
+    if (isnan(real(lambdaL))) then
+      lambdaL = zero
+    end if
 
     ! Finally, communicate the step size across processes and return
-    call mpi_allreduce(lambdaL, lambdaP, 1_intType, adflow_real, &
+    ! mpi allreduce is not defined for complex numbers with the min operation
+    ! so we will use the tmp variable to receive
+    call mpi_allreduce(lambdaL, tmp, 1_intType, MPI_DOUBLE, &
          mpi_min, ADflow_comm_world, ierr)
     call EChk(ierr,__FILE__,__LINE__)
 
+#ifndef USE_COMPLEX
+    lambdaP = tmp
+#else
+    ! finally, as a safety check, purge the complex part of lambda
+    lambdaP = cmplx(tmp, 0.0_realType)
```

Comment:
So lambdaP is always real?
lambdaL is always real

Reply:
Shouldn't all lambda's always be real

Reply:
if all lambdas are always real, then you could probably get rid of your `tmp` variables

Reply:
I needed the `tmp` because the MPI_MIN operation in the MPI_Allreduce is not defined for complex numbers. It just fails. So I have to use an always real type to do the allreduce.

Reply:
I actually had a comment right above as well:
```
! mpi allreduce is not defined for complex numbers with the min operation
! so we will use the lambdaP_recv variable to receive
```

Reply:
oh I just realized, `lambdaP` is not `alwaysReal`, it is just `realType`. As a result, I needed that additional always real variable to receive the MPI allreduce's output